src/pre-engagement-form/form-components/form-component.tsx

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+/**
+ * This is a helper wrapper component.
+ * It handles things that are common to all form-components:
+ * - React Hook Form Binding
+ * - Validation and Error Messaging
+ * - Component Label
+ */
+
+/* eslint-disable react/require-default-props */
+import React, { useRef } from 'react';
+import { connect } from 'react-redux';
+import { useFormContext, Controller, UseControllerProps, FieldError } from 'react-hook-form';
+
+import { useLocalization } from '../localization';
+import { setValue } from '../state';
+import { Label } from './styles';
+
+type OwnProps = {
+  label: string;
+  defaultValue?: string;
+  children: JSX.Element;
+};
+
+type Props = OwnProps & UseControllerProps & typeof mapDispatchToProps;
+
+// TODO: Displaying error can be more elegant and robust
+const getErrorMessage = (error: any | FieldError | undefined, strings: Record<string, string>) => {
+  if (error === undefined) {
+    return '';
+  }
+
+  if (error.type === 'required') {
+    return error.message || strings.FieldValidationRequiredField;
+  }
+
+  if (error.type === 'pattern') {
+    /**
+     * TODO: ideally there should be a a more generic error string like
+     * "FieldValidationInvalidPattern". Right now, Email Validation is the only
+     * validation we're using besides Required Fields, so we're fine with
+     * falling back to "FieldValidationInvalidEmail".
+     */
+    return error.message || strings.FieldValidationInvalidEmail;
+  }
+
+  return '';
+};
+
+const FormComponent: React.FC<Props> = ({ name, label, rules, defaultValue, handleChange, children }) => {
+  const { strings, getLabel } = useLocalization();
+  const {
+    control,
+    formState: { errors },
+  } = useFormContext();
+  const inputRef = useRef<HTMLInputElement | HTMLSelectElement | null>(null);
+  const isRequired = Boolean(rules?.required);
+
+  return (
+    <Label htmlFor={name}>
+      <span className="label">
+        {getLabel(label)} {isRequired && '*'}
+      </span>
+      <Controller
```

Comment:
Just out of curiosity, why are you using the `Controller` rather than the passing the `ref` and handlers directly to the input? DRYing things?

Reply:
It ended up being a little more DRY, but that was not the main reason at all 😂

Unfortunately WebChat's exposed `styled` function is **different** than FlexUI's `styled`, the type of the generated object is different, which is so sad. I was having a lot issues in trying to use refs directly in WebChat, I could only make it work using RHF's `<Controller>`. That's the real reason xD